client/components/NavBar.js

```diff
@@ -1,58 +1,122 @@
+                  Order History
+                </Link>
+                { user.userType === 'admin' ? 
+                  <Link to="/admin/pending" className="navbar-item">Orders to Ship</Link>
+                  : null }
             </div>
-
-            <div class="navbar-end">
-              <div class="navbar-item">
-                <div class="buttons">
-                  <a class="button is-black">
-                    <strong>Register</strong></a>
-                  <a class="button is-black">Log in</a>
-                </div>
-              </div>
+            <div className= "topnav-right">  
+              <span>Welcome, {user.id ? user.firstName : 'Guest'}!</span>
+                {user.id ? (
+                  <div className="buttons">
+                    <Link to="/orderHistory" className="button is-black">Order History</Link>
+                    <Link to="/login" className="button is-black" onClick={logout}>Log out</Link>
+                  </div>
+                )
+                  : (
+                    <div className="buttons">
+                      <Link to="/createUser" className="button is-black">Register</Link>
+                      <Link to="/login" className="button is-black">Log in</Link>
+                    </div>
+              )}
             </div>
           </div>
-        </nav>
-
-        <div>
-          <div class="sidenav">
-            <a href="#">All Costumes</a>
-            <a href="#">Men</a>
-            <a href="#">Women</a>
-            <a href="#">Kids</a>
-            <a href="#">Scary</a>
-            <a href="#">Funny</a>
-            <a href="#">Animals</a>
-            <a href="#">Movies</a>
-          </div>
+
+        <div className="sidenav" role="navigation" aria-label="search costumes by category">
+          <p>Select a Category:</p>
+          <Link className={selectedCategory === "all" ? "selected" : null} to="/categories/all" onClick={() => selectCategory('all')}>All</Link>
+          {
+            categories.map((category) => (
+              <Link className={selectedCategory === category.title ? "selected" : null} key={category.id} 
+              to={`/categories/${category.title}`} onClick={() => selectCategory(`${category.title}`)}>
+                {category.title}
+              </Link>
+            ))
+          }
         </div>
       </div>
-    )
+    );
   }
 }
+
+const mapStateToProps = (state, ownProps) => ({
+  categories: state.categories,
+  user: state.user,
+  // selectedCategoryTitle: ownProps.match.params.categoryTitle ? ownProps.match.params.categoryTitle : 'all',
+  selectedCategoryTitle: 'Animals',
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  fetchCategories: () => {
+    dispatch(fetchCategories());
+  },
+  logout: () => {
+    dispatch(logout());
+  },
+  getUser: () => {
+    dispatch(getUser());
+  },
+  checkCookiesSetSession: () => {
+    dispatch(checkCookiesSetSession());
```

Comment:
If we add a `return` we're going to receive the return value of `checkCookiesSetSession()` - but I'm not sure thats a `Promise` either...